### PR TITLE
VMHostPermission DSC Resource

### PR DESCRIPTION
**Pull request by SimeonGerginov**
### Added
- Added VMHostPermission DSC Resource.
- Added Unit Tests for VMHostPermission DSC Resource.
- Added Integration Tests for VMHostPermission DSC Resource.
- Added Documentation and example Configurations for VMHostPermission DSC Resource.

### CODE REVIEW

**Comment by SimeonGerginov** [comment]
Example usage of VMHostPermission DSC Resource:

```
VMHostPermission VMHostPermission {
    Server = $Server
    Credential = $Credential
    EntityName = 'MyDscVM'
    EntityLocation = 'MyDscResourcePool/MyDscvApp'
    EntityType = 'VM'
    PrincipalName = 'MyDscPrincipal'
    RoleName = 'MyDscRole'
    Ensure = 'Present'
    Propagate = $true
}
```

File: Source/VMware.vSphereDSC/DSCResources/VMHostPermission.ps1 [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the Root Resource Pool from the VMHost.
+    #>
+    [PSObject] GetRootResourcePool() {
+        try {
+            $resourcePoolsType = (Get-ResourcePool -Server $this.Connection | Select-Object -First 1).GetType()
+            $rootResourcePool = Get-Inventory -Server $this.Connection -ErrorAction Stop -Verbose:$false |
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/DSCResources/VMHostPermission.ps1]: Check whether you can replace Get-Inventory with Get-ResourcePool

File: Source/VMware.vSphereDSC/DSCResources/VMHostPermission.ps1 [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+    If Ensure is 'Present' and the Principal is not found, an exception is thrown. If Ensure is 'Absent' and the Principal is not found, $null is returned.
+    #>
+    [PSObject] GetVMHostPrincipal() {
+        $getVIAccountParams = @{
+            Server = $this.Connection
+            Verbose = $false
+        }
+
+        # If the Principal is a Domain User, we should extact the Domain and User names from the Principal name.
+        if ($this.PrincipalName -Match '\\') {
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/DSCResources/VMHostPermission.ps1]: Check domain\user is always the case. I guess user@domain is also a valid domain user format.

File: Source/VMware.vSphereDSC/DSCResources/VMHostPermission.ps1 (hunk 1 above)
Comment (SimeonGerginov) [reply Source/VMware.vSphereDSC/DSCResources/VMHostPermission.ps1]: Done.

File: Source/VMware.vSphereDSC/DSCResources/VMHostPermission.ps1 (hunk 2 above)
Comment (SimeonGerginov) [reply Source/VMware.vSphereDSC/DSCResources/VMHostPermission.ps1]: Done.

**Pull request merged**